### PR TITLE
Fixes for swift

### PR DESCRIPTION
**Pull request by justinboswell**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/docker-images/swift-5-centos-x64/Dockerfile [hunk 1]
```diff
@@ -10,34 +10,11 @@ RUN yum -y update \
...
-    cmake \
-    cmake3 \
     openssl-devel \
-    gcc-c++ \
     && yum clean all \
-    && rm -rf /var/cache/yum 
-###############################################################################
-# Handle cmake on centos properly
-###############################################################################
-RUN sudo alternatives --install /usr/local/bin/cmake cmake /usr/bin/cmake 10 \
```
Comment (kneekey23) [inline .github/docker-images/swift-5-centos-x64/Dockerfile:25]: wasnt sure if any of this was needed for the builder, but glad to see it's not :)

File: builder/actions/install.py [hunk 2]
```diff
@@ -99,16 +99,19 @@ def export_compiler(compiler, env):
...
-    cxx_path = toolchain.cxx_compiler_path()
-    if cc_path:
-        env.shell.setenv('CC', cc_path)
-    else:
-        print('WARNING: C compiler {} could not be found for export'.format(compiler))
-    if cxx_path:
-        env.shell.setenv('CXX', cc_path)
-    else:
-        print('WARNING: CXX compiler {} could not be found for export'.format(compiler))
+    if not env.shell.getenv('CC'):
```
Comment (justinboswell) [inline builder/actions/install.py:102]: We should never override the environment if it's specified. Or should we check to see if `--compiler` was specified on the command line and override it then?

File: .github/docker-images/swift-5-centos-x64/Dockerfile [hunk 3]
```diff
@@ -5,39 +5,15 @@ FROM swift:5.4-centos7
 # Install prereqs
 # any prereqs that appear to be missing are installed on base swift image i.e. tar, git
 ###############################################################################
-RUN yum -y update \
```
Comment (justinboswell) [inline .github/docker-images/swift-5-centos-x64/Dockerfile:8]: Performing this update on centos was breaking glibc's headers, it appears to actually downgrade them.

**Review by kneekey23 — APPROVED** [review]
LGTM

**Pull request merged**